### ISSUE TITLE
[Dev] Include asset makers into CI tests

### ISSUE THREAD

**Issue by DanielYang59** [janosh/pymatviz#164]
Should include asset maker ipynb into CI tests.
Link to #62.

**Comment by janosh** [comment]
i briefly considered adding this in https://github.com/janosh/pymatviz/pull/163 via a new workflow file `test-make-assets.yml` but it would be slow and a waste of resources to run all these scripts on every commit and merge. ideally, the scripts should only when any of the code they rely on changed. e.g. `make_assets/scatter.py` would only run if it changed itself or if `scatter.py` changed. it's possible to check what changed with `git diff`. of course, the script could break due to changes in other files as well but I think the risk of that is not high enough to rerun the script every time

```yml
name: Test asset scripts

on:
  pull_request:
    branches: [main]
  push:
    branches: [main]
  workflow_dispatch:

jobs:
  run:
    runs-on: ubuntu-latest
    steps:
      - name: Check out repository
        uses: actions/checkout@v4

      - name: Set up Python
        uses: actions/setup-python@v5
        with:
          python-version: 3.9

      - name: Install package and dependencies
        run: pip install -e .

      - name: Run all make_assets scripts concurrently
        run: |
          scripts=$(find examples/make_assets -name '*.py')
          exit_code=0

          for script in $scripts; do
            echo "Running $script"
            python "$script" &
          done

          for job in $(jobs -p); do
            wait $job || let "exit_code+=1"
          done

          if [ "$exit_code" != "0" ]; then
            echo "One or more scripts failed. Please check the logs above for details."
            exit $exit_code
          fi
```

**Comment by DanielYang59** [comment]
Would be fixed by #195, thanks @janosh
